src/modules/PowerShell/FinOpsHub/FinOpsHub.psm1

```diff
@@ -0,0 +1,193 @@
+data LocalizedData
+{
+    # culture="en-US"
+    ConvertFrom-StringData -StringData @'
+    DeployResourceGroupScope = Deploying hub from template '{0}' to resource group '{1}'.
+    DeploySubscriptionScope = Deploying hub from template '{0}' to subscription '{1}'.
+    DeployTenantScope = Deploying hub from template '{0}' to tenant '{1}'.
+    NotLoggedIn = Not logged into Azure. Launching 'Login-AzAccount'.
+    LoggedIn = Currently logged into subscription '{0}' '{1}'.
+'@
+}
+
+<#
+    .SYNOPSIS
+        Gets Azure context for logged in user. Will prompt for login if not currently logged in.
+#>
+function Assert-AzContext
+{
+    [CmdletBinding()]
+    param()
+    $context = Get-AzContext
+    if ($null -eq $context)
+    {
+        Write-Warning -Message ($LocalizedData.NotLoggedIn)
+        $null = Login-AzAccount
+        $context = Get-AzContext
+    }
+
+    Write-Verbose -Message ($LocalizedData.LoggedIn -f $context.Subscription.Name, $context.Subscription.Id)
+    return $context
+}
+
+<#
+    .SYNOPSIS
+        Deploys a FinOpsHub instance.
+
+    .PARAMETER Path
+        Path to the .bicep template.
+
+    .PARAMETER HubName
+        Name of the FinOpsHub instance.
+
+    .PARAMETER Scope
+        Scope to deploy FinOpsHub to, either ResourceGroup, Subscription, or Tenant.
+
+    .PARAMETER ResourceGroupName
+        Name of the resource group to deploy to. Will be created if it doesn't exist. Only used for ResourceGroup scope.
+
+    .PARAMETER Location
+        Azure location to execute the deployment from.
+
+    .PARAMETER StorageSku
+        Optional. Storage account SKU. LRS = Lowest cost, ZRS = High availability. Note Standard SKUs are not available for Data Lake gen2 storage.
+
+    .PARAMETER Tags
+        Optional. Tags for all resources.
+
+    .PARAMETER ExportScopes
+        Optional. List of scope IDs to create exports for.
+
+    .EXAMPLE
+        Deploy-FinOpsHub -Path c:\finopshub\main.bicep -HubName MyHub -ResourceGroupName MyExistingResourceGroup -Location westus -Scope ResourceGroup
+        
+        Deploys a new FinOps hub instance named MyHub to an existing resource group name MyExistingResourceGroup.
+    
+    .EXAMPLE
+        Deploy-FinOpsHub -Path c:\finopshub\main.bicep -HubName MyHub -Location westus -Scope Subscription
+        
+        Deploys a new FinOps hub instance named MyHub to your currently logged into Azure subscription.
+
+    .EXAMPLE
+        Deploy-FinOpsHub -Path c:\finopshub\main.bicep -HubName MyHub -Location westus -Scope Tenant
+        
+        Deploys a new FinOps hub instance named MyHub to your currently logged into Azure tenant.
+#>
+function Deploy-FinOpsHub
+{
+    [CmdletBinding()]
+    param
+    (
+        [Parameter(Mandatory = $true)]
+        [ValidateScript({Test-Path -Path $_})]
+        [ValidateScript({{[System.IO.Path]::GetExtension($_) -eq '.bicep'}})]
+        [string]
+        $Path,
```

Comment:
Given this is a published module, should we build the bicep directly into the PSM or pull it from GitHub ourselves? Why do they need to tell us what Bicep file to deploy? They can just tell us what version to deploy. Thoughts?

Reply:
I like the idea of pulling it directly from a repository for sure.

Reply:
Changed Path to HubVersion. Right now it can only pull from Releases. If we need to support raw content I need to figure out which files are needed to deploy the hub still. Figured I would wait until we had a deployable hub to worry about that though.